tests/Unit/Adapter/ValidateTest.php

```diff
@@ -33,12 +33,28 @@
 
 class ValidateTest extends TestCase
 {
+    /**
+     * @var Validate
+     */
+    private $validate;
+
+    /**
+     * @param string $name
+     * @param string $dataName
+     */
+    public function __construct($name = null, array $data = [], $dataName = '')
+    {
+        parent::__construct($name, $data, $dataName);
+
+        $this->validate = new Validate();
+    }
+
     /**
      * @dataProvider getIsOrderWay
      */
     public function testIsOrderWay(int $expected, $input): void
     {
-        self::assertEquals($expected, Validate::isOrderWay($input));
+        self::assertEquals($expected, $this->validate::isOrderWay($input));
```

Comment:
Hi! I'm sure you did this on purpose but I don't understand why.

Why
```
self::assertEquals($expected, $this->validate::isOrderWay($input));
```
is better than
```
self::assertEquals($expected, Validate::isOrderWay($input));
```

Reply:
@matks `Validate` has been injected in constructor. To keep consistent we need to use this one

Reply:
Oh! I understand now. The root issue is that adapter Validate class still using static calls, that directly relate to unnamespaced class.

What a very weird class `PrestaShop\PrestaShop\Adapter\Validate` 😅